src/components/views/rooms/MessageContextMenu.js

```diff
@@ -57,25 +57,42 @@ module.exports = React.createClass({
         if (this.props.onFinished) this.props.onFinished();
     },
 
+    onCancelSendClick: function() {
+        Resend.removeFromQueue(this.props.mxEvent);
+        if (this.props.onFinished) this.props.onFinished();
```

Comment:
Looks like this just stops retries for sending the message and doesn't cancel any requests that are currently in progress? Granted you'll never be able to guarantee that the message hadn't already been delivered, but this could be confusing if you cancel a message when you lose connectivity, then get connectivity back and find that it appears in the timeline because of normal tcp level retries.